pkg/common-controller/groupsnapshot_controller_helper.go

```diff
@@ -724,10 +737,10 @@ func (ctrl *csiSnapshotCommonController) bindandUpdateVolumeGroupSnapshot(groupS
 
 // createGroupSnapshotContent will only be called for dynamic provisioning
 func (ctrl *csiSnapshotCommonController) createGroupSnapshotContent(groupSnapshot *crdv1alpha1.VolumeGroupSnapshot) (*crdv1alpha1.VolumeGroupSnapshotContent, error) {
-	klog.Infof("createSnapshotContent: Creating group snapshot content for groupn snapshot %s through the plugin ...", utils.GroupSnapshotKey(groupSnapshot))
+	klog.Infof("createSnapshotContent: Creating group snapshot content for group snapshot %s through the plugin ...", utils.GroupSnapshotKey(groupSnapshot))
 
 	/*
-		TODO: Add finalizer to group snapshot
+		TODO: Add PVC finalizer
```

Comment:
Why do we need PVC finalizer?

Reply:
This is an open question I had - do we add any finalisers to PVCs, similar to the case for individual VolumeSnapshots?

Reply:
Yes, we do need to add PVC finalizers when creating a snapshot from it.

Reply:
So will the individual snapshot creation handle that, or does the group snapshot also need to add a finaliser?

Reply:
Since the creation of the individual VolumeSnapshot API objects are handled by the group snapshot controller, the PVC finalizer should also be added by the group snapshot controller. Let me know if you see any problems.

Reply:
In the first revision, let's try to separate the logic as much as possible to avoid introducing regression to the existing volume snapshot feature.

Reply:
I'm going to merge this PR. Please address this comment in a followup PR.